src/test/java/org/folio/fqm/ModFqmManagerIT.java

```diff
@@ -0,0 +1,127 @@
+
+/**
+ * Check the shaded fat uber jar and Dockerfile:
+ *
+ * <p>Test /admin/health.
+ *
+ * <p>Test installation and migration with smoke test.
+ */
+@Testcontainers
+class ModFqmManagerIT {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ModFqmManagerIT.class);
+  private static final boolean IS_LOG_ENABLED = false;  // for debugging set to true!
+  private static final Network network = Network.newNetwork();
+
+  @Container
+  public static final PostgreSQLContainer<?> postgres =
+    new PostgreSQLContainer<>("postgres:12-alpine")
+      .withNetwork(network)
+      .withNetworkAliases("mypostgres")
+      .withExposedPorts(5432)
+      .withUsername("myusername")
+      .withPassword("mypassword")
+      .withDatabaseName("mypostgres");
+
+  @Container
+  public static final GenericContainer<?> module =
+    new GenericContainer<>(
+      new ImageFromDockerfile("mod-fqm-manager").withFileFromPath(".", Path.of(".")))
+      .withNetwork(network)
+      .withExposedPorts(8081)
+      .withEnv("DB_HOST", "mypostgres")
+      .withEnv("DB_PORT", "5432")
+      .withEnv("DB_USERNAME", "myusername")
+      .withEnv("DB_PASSWORD", "mypassword")
+      .withEnv("DB_DATABASE", "mypostgres")
+      .dependsOn(postgres);
+
+  @BeforeAll
+  public static void beforeAll() {
+    RestAssured.reset();
+    RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
+    RestAssured.baseURI = "http://" + module.getHost() + ":" + module.getFirstMappedPort();
+    if (IS_LOG_ENABLED) {
+      module.followOutput(new Slf4jLogConsumer(LOG).withSeparateOutputStreams());
+    }
+  }
+
+  @BeforeEach
+  public void beforeEach() {
+    RestAssured.requestSpecification = null;
+  }
+
+  @Test
+  public void health() {
+    // request without X-Okapi-Tenant
+    when()
+      .get("/admin/health")
+      .then()
+      .statusCode(200)
+      .body(containsString("\"UP\""));
+  }
+
+  @Test
+  public void installAndUpgrade() {
+    String body = """
+                  {"module_to": "999999.0.0",
+                   "parameters": [{"key": "loadReference", "value": "true"},
+                                  {"key": "loadSample",    "value": "true"} ]
+                  }
+                  """;
+    postTenant(body);
+
+    // migrate from 0.0.0, migration should be idempotent
+    body = """
+           {"module_from": "0.0.0",
+           """ + body.substring(1);
+    postTenant(body);
+
+    smokeTest();
+  }
+
+  private void postTenant(String body) {
+    given()
+      .header("X-Okapi-Tenant", "beeuni")
+      .contentType("application/json")
+      .body(body)
+      .when()
+      .post("/_/tenant")
+      .then()
+      .statusCode(204);
+  }
+
+  private void smokeTest() {
+    given()
+      .header("X-Okapi-Tenant", "beeuni")
+      .contentType("application/json")
+      .when()
+      .get("/entity-types")
+      .then()
+      .statusCode(200)
+      .body("$.size()", is(3));
```

Comment:
The 3 seems a little fragile, since it's so specific (every time we add a new entity type, this test will need updated). Maybe `.body("$.size()", greaterThanOrEqualTo(3));`?

Reply:
Fair enough. Eventually, we'll probably want to figure out a nicer way to get the number of entity types (maybe a `select count(*) from entity_types` directly on the container from the test or something), but I'm okay with manually updating it for now.